File: Using Java/350. Intersection of Two Arrays II.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        
        List<Integer> lst = new ArrayList<>();
        int i = 0;
        int j = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j]) {
                i++;
            } else if (nums1[i] > nums2[j]) {
                j++;
            } else {
                lst.add(nums1[i]);
                i++;
                j++;
            }
        }
        
        int[] result = new int[lst.size()];
        for (int k = 0; k < lst.size(); k++) {
            result[k] = lst.get(k);
        }
        
        return result;
    }
}

File: using python/350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1, nums2):
        nums1.sort()
        nums2.sort()
        lst = []
        i = j = 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                i += 1
            elif nums1[i] > nums2[j]:
                j += 1
            else:
                lst.append(nums1[i])
                i += 1
                j += 1
        return lst
      
#to run this code locally
# myclass=Solution()
# n1 = [1,2,2,1]
# n2 = [2,2]
# myclass.intersect(n1,n2)
